services/transfer_service.py

```
@@ -582,30 +899,115 @@ def cancel_transfer(self, transfer_id: str) -> bool:
+            'eta_seconds': None
+        })
+
+        if has_process:
             try:
                 import psutil
-                process = psutil.Process(transfer['rsync_process_id'])
-                process.terminate()
-                
-                # Update status
+                psutil.Process(pid).terminate()
+            except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)